examples/bri-3/src/bri/workgroup/workgroups/agents/workgroups.agent.ts

```diff
@@ -63,6 +70,27 @@ export class WorkgroupAgent {
     workgroupToUpdate.updateParticipants(participants);
   }
 
+  public async fetchArchiveCandidateAndThrowIfArchiveValidationFails(
+    id: string,
+  ): Promise<Workgroup> {
+    const workgroupToArchive =
+      await this.workgroupStorageAgent.getWorkgroupById(id);
+
+    if (!workgroupToArchive) {
+      throw new NotFoundException(WORKGROUP_NOT_FOUND_ERR_MESSAGE);
+    }
+
+    if (workgroupToArchive.status != WorkgroupStatus.ACTIVE) {
```

Comment:
Any cases where you would favor making more unique prisma queries to the DB in favor of multiple checks on the agent level? For example, adding something in prisma like Where: status == ACTIVE. Just curious to everyone opinion. Not suggesting changing this. @ognjenkurtic @skosito @Ybittan @Manik-Jain @biscuitdey

Reply:
you think to use same query to check if active workgroup with given id exists? maybe we can add it to getWorkgroupById, because there should not be a reason to get workgroup if its not active anymore
but for now we can keep it like this too i think